### DISCUSSION TITLE
emailjs can't find path for attachment files

### DISCUSSION THREAD

**Discussion by RytisTam** [eleith/emailjs#335] [ideas / feature requests]
Hi. I can't seem to figure out why, but whenever I try to attach a file while sending email, I get the error:
`Error: image.jpeg does not exist`

I'm using nextJs with emailjs, and I have a file image.jpeg in /public/image.jpeg, I've tried these combinations:

- /image.jpeg
- /public/image.jpeg
- localhost:3000/image.jpeg
- Also tried to pull one off of internet, just: www.url.com/images/image.jpeg

None of these seem to work

I'm constructing my message this way:
`const message = {
    ...
    attachment: [
      {data: `<html>...</html>`, alternative: true},
        { path: '/image.jpeg', type: 'image/jpeg', name: 'image.jpeg' },
    ],
};
`

What am I missing here? Thanks.

**Comment by eleith** [comment]
the error is coming from https://github.com/eleith/emailjs/blob/8b3c0b16f81e9ab2a3ca518cd893b3474f87cda6/smtp/message.ts#L230

likely your path to the image is not aligned with the directory your process is running from.

you can look at one of our tests to see how we approach image embedding

https://github.com/eleith/emailjs/blob/main/test/message.ts#L218

**Comment by RytisTam** [comment]
Yes, sorry, my bad.

Turns out this has nothing to do with emailjs but rather with NextJS. I'm using emailjs in NextJS's /page/api folder and I guess it handles paths a little different therefore I needed to use absolute path. (No idea if that's the best solution)

**Thank you for your reply tho!**
